pkg/diff/diff.go

```diff
@@ -0,0 +1,691 @@
+			o.new = newR.Commit
+			rawRepos[name] = o
+			continue
+		}
+
+		// new side only
+		g := gitRepo{
+			name:    name,
+			url:     newR.URL,
+			dir:     filepath.Join(opts.dir, name),
+			new:     newR.Commit,
+			remotes: make(map[string]remote),
+			filter:  opts.filter,
+			diffAlg: extracted.diffAlg,
+		}
+		log.Printf("[INFO] Working directory for raw: %s", g.dir)
+		if err := g.prepare(); err != nil {
+			return nil, errors.Wrapf(err, "init raw git repo. URL: %s", g.url)
+		}
+		rawRepos[name] = g
+	}
+
+	return rawRepos, nil
+}
+
+func listRawfiles(extracted gitRepo, oldPath, newPath string) ([]rawfile, error) {
+	bs, err := extracted.showBlob(extracted.old, oldPath)
+	if err != nil {
+		return nil, errors.Wrapf(err, "git show. commit: %s, path: %s", extracted.old, oldPath)
+	}
+	var oldData dataTypes.Data
+	if err := json.Unmarshal([]byte(bs), &oldData); err != nil {
+		// Don't report error and just continue because older Data type does not include raw file paths.
+		// If dataTypes.Data struct will change, it'll be time to consider unmarhsalling it with interface{}.
+		log.Printf("[WARN] unmarshal of %s failed, probably data type is stale", oldPath)
+		return nil, nil
+	}
+
+	bs, err = extracted.showBlob(extracted.new, newPath)
+	if err != nil {
+		return nil, errors.Wrapf(err, "git show. commit: %s, path: %s", extracted.new, newPath)
+	}
+	var newData dataTypes.Data
+	if err := json.Unmarshal([]byte(bs), &newData); err != nil {
+		log.Printf("[WARN] unmarshal of %s failed, probably data type is stale", newPath)
+		return nil, nil
+	}
+
+	var rawfiles = make(map[string]rawfile)
+	for _, fullpath := range oldData.DataSource.Raws {
+		ss := strings.Split(fullpath, string(filepath.Separator))
+		if len(ss) == 0 {
+			return nil, errors.Errorf("raw path does not contain repository name. path: %s", fullpath)
+		}
+
+		rawfiles[fullpath] = rawfile{
+			repo: ss[0],
+			path: strings.Join(ss[1:], string(filepath.Separator)),
+			old:  true,
+		}
+	}
+	for _, fullpath := range newData.DataSource.Raws {
+		ss := strings.Split(fullpath, string(filepath.Separator))
+		if len(ss) == 0 {
+			return nil, errors.Errorf("raw path does not contain repository name. path: %s", fullpath)
+		}
+
+		f := rawfiles[fullpath]
+		rawfiles[fullpath] = rawfile{
+			repo: ss[0],
+			path: strings.Join(ss[1:], string(filepath.Separator)),
+			old:  f.old,
+			new:  true,
+		}
+	}
+
+	return slices.Collect(maps.Values(rawfiles)), nil
+}
+
+type gitRepo struct {
+	name             string
+	url              string
+	dir              string
+	old              string
+	new              string
+	remotes          map[string]remote
+	filter           string
+	diffAlg          string
+	archiveExhausted bool
+}
+
+type remote struct {
+	name         string
+	url          string
+	archiveIndex int
+	partial      bool
+}
+
+func (g *gitRepo) prepare() error {
+	if _, err := os.Stat(g.dir); err != nil {
+		if !os.IsNotExist(err) {
```

Comment:
> This function predates [errors.Is](https://pkg.go.dev/errors#Is). It only supports errors returned by the os package. New code should use errors.Is(err, fs.ErrNotExist).

https://pkg.go.dev/os#IsNotExist

```suggestion
		if !errors.Is(err, fs.ErrNotExist) {
```